flexget/components/notify/notifiers/mqtt.py

```diff
@@ -0,0 +1,185 @@
+            client.tls_set(ca_certs=certs.get('broker_ca_cert'),
+                          certfile=None,
+                          keyfile=None,
+                          cert_reqs=ssl.CERT_NONE)
+
+            client.tls_insecure_set(True)
+
+            logger.verbose('Basic SSL/TLS encrypted communications enabled')
+            logger.verbose('TLS insecure cert mode enabled. Broker cert will not be validated')
+
+
+        #Handle SSL/TLS communication with certificate authentication
+        if config.get('certificates',False):
+            certs = config.get('certificates',{})
+            logger.debug('TLS certificate config: {}',str(certs))
+
+            tls_version_map = {'tlsv1.2': ssl.PROTOCOL_TLSv1_2, 'tlsv1.1': ssl.PROTOCOL_TLSv1_1, 'tlsv1': ssl.PROTOCOL_TLSv1, '': None}
+            tls_version = tls_version_map.get(certs.get('tls_version'),ssl.PROTOCOL_TLSv1_2)
+            logger.verbose('TLS version is {}',str(tls_version))
+
+            cert_required = ssl.CERT_REQUIRED if certs.get('validate_broker_cert', True) else ssl.CERT_NONE
+
+            client.tls_set(ca_certs=certs.get('broker_ca_cert'),
+                          certfile=certs.get('client_cert'),
+                          keyfile=certs.get('client_key'),
+                          cert_reqs=cert_required,
+                          tls_version=tls_version)
+
+            if not certs.get('validate_broker_cert'):
+                client.tls_insecure_set(True)
+                logger.debug('TLS insecure cert mode enabled. Broker cert will not be validated')
+            else:
+                logger.debug('TLS secure cert mode enabled. Broker cert will be validated')
+
+        #Handle user/pass authentication
+        if config.get('username',False) or config.get('password',False):
+            logger.debug('Credential passwords s are redacted to protect the innocent...')
+            logger.debug('Auth credentials: username=[{}] password sha256 hash is "{}"',config.get('username'),sha256(str(config.get('password')).encode('utf-8')).hexdigest())
+            logger.debug('You can validate them yourself by calculating the sha256 hex digest of your password string (google is your friend if you do not know how to do this)')
+            logger.debug('Note: a password that is not provided (i.e. None) will hash to "{}"',sha256(str(None).encode('utf-8')).hexdigest())
+
+            client.username_pw_set=(config.get('username'),config.get('password'))
+
+        try:
+            logger.verbose("Connecting to {}:{}",config.get('broker_address'),str(config.get('broker_port')))
+            client.connect(config.get('broker_address'), config.get('broker_port'), config.get('broker_timeout'))
+            logger.verbose("Connected to MQTT broker")
+        except Exception as e:
+            raise PluginWarning('Error connecting to MQTT broker:  %s' % e)
```

Comment:
syntax of this (and other) exceptions can be fstring:
```suggestion
            raise PluginWarning(f'Error connecting to MQTT broker: {e}')
```

Reply:
Addressed with commit 5b23add